ratchet-tree.go

```diff
@@ -287,31 +332,23 @@ func (t *RatchetTree) Decap(from leafIndex, context []byte, path *DirectPath) []
 			if !temp.PublicKey.equals(&pathNode.PublicKey) {
 				panic(fmt.Errorf("mls:rtn: incorrect public key"))
 			}
-			t.Nodes[dp[i+1]].merge(temp)
+			t.Nodes[dp[i+1]].mergePrivate(temp.PrivateKey)
 		} else {
-			t.Nodes[dp[i+1]].merge(&RatchetTreeNode{PublicKey: &pathNode.PublicKey})
+			t.Nodes[dp[i+1]].mergePublic(&pathNode.PublicKey)
 		}
```

Comment:
@bifurcation  love the split of merge() into private and public ..   makes the context so much clearer